editor/src/core/model/performance-scripts.tsx

```diff
@@ -331,6 +339,197 @@ export function useTriggerSelectionPerformanceTest(): () => void {
   return trigger
 }
 
+export function useTriggerAbsoluteMovePerformanceTest(): () => void {
+  const dispatch = useEditorState(
+    React.useCallback((store) => store.dispatch as DebugDispatch, []),
+    'useTriggerAbsoluteMovePerformanceTest dispatch',
+  )
+  const allPaths = useRefEditorState(
+    React.useCallback((store) => store.derived.navigatorTargets, []),
+  )
+  const metadata = useRefEditorState(React.useCallback((store) => store.editor.jsxMetadata, []))
+  const selectedViews = useRefEditorState(
+    React.useCallback((store) => store.editor.selectedViews, []),
+  )
+  const trigger = React.useCallback(async () => {
+    const initialTargetPath = [...allPaths.current].sort(
+      (a, b) => EP.toString(b).length - EP.toString(a).length,
+    )[0]
+    // This is very particularly tied to the test project, we _really_ need to pick the
+    // right element because our changes can cause other elements to end up on top of the
+    // target we want.
+    const parentParentPath = EP.parentPath(EP.parentPath(initialTargetPath))
+    const grandChildrenPaths = allPaths.current.filter((path) => {
+      return EP.pathsEqual(parentParentPath, EP.parentPath(EP.parentPath(path)))
+    })
+    if (grandChildrenPaths.length === 0) {
+      console.info('ABSOLUTE_MOVE_TEST_ERROR')
+      return
+    }
+    const targetPath = forceNotNull('Invalid array.', last(grandChildrenPaths))
+
+    // Switch Canvas Strategies on.
+    const strategiesCurrentlyEnabled = isFeatureEnabled('Canvas Strategies')
+    setFeatureEnabled('Canvas Strategies', true)
+    // Delete the other children that just get in the way.
+    const parentPath = EP.parentPath(targetPath)
+    const siblingPaths = allPaths.current.filter(
+      (path) => EP.isChildOf(path, parentPath) && !EP.pathsEqual(path, targetPath),
+    )
+    await dispatch(
+      siblingPaths.map((path) => deleteView(path)),
+      'everyone',
+    ).entireUpdateFinished
+    // Focus the target so that we can edit the child div inside it.
```

Comment:
what if I don't have a focusable component in my test project? will this cause a problem?

Reply:
Yes.

Reply:
Unfortunately right now this test is very specific to the current performance test project, which does seem to imply that either there's an issue with applying the strategy directly on any element (even after giving it the required style props), or that there's an issue with firing the required mouse events to invoke the strategy directly on a specific element. I suggest we look to address that separately in some follow up work, perhaps after we have had time to evaluate this testing strategy (pun partially intended) on a number of strategies.

Reply:
I think the ideal case would be to pair a testing strategy with a project definition. Especially if it was to be done in such a way that didn't involve the saving and the shenanigans it brings in as well.

Reply:
All of the existing tests are brittle like this to some extent, but since this test is for absolutely positioned elements and our current test project of choice is entirely done with flex this one is just that bit more troublesome.

Reply:
> I think the ideal case would be to pair a performance test with a project definition. Especially if it was to be done in such a way that didn't involve the saving and the shenanigans it brings in as well.

Hmmm I really like this idea, but that definitely makes testing with puppeteer a bit awkward (because we'd either have to run multiple separate tests like that, or trigger page navigations after each test). I don't think that's entirely implausible though.

Alternatively, we could investigate a way to designate the target elements of each test in a given project (maybe via a `data-label` prop on the element for a specific test, or by enabling staging to retain UIDs via a feature flag)?

Reply:
I say that as well given that there's already interest in testing with different projects, which as-is would be very clunky as we're having to make sure that projects exist in each environment its being run in.

Reply:
we could store the test projects in the editor code, and clicking the test button could start by loading the project contents and runnin the test. that would ensure we really are testing the same thing everywhere

Reply:
okay I don't think this is a blocker issue, but we should have a follow up TODO about making the tests more portable 👍